go/ngql/types.go

```diff
@@ -257,63 +257,80 @@ func getListValues(v types.Value, args map[string]interface{}) (interface{}, err
 }
 
 var setArgs = graphql.FieldConfigArgument{
+	atKey:    &graphql.ArgumentConfig{Type: graphql.Int},
 	countKey: &graphql.ArgumentConfig{Type: graphql.Int},
 }
 
 func getSetValues(v types.Value, args map[string]interface{}) (interface{}, error) {
 	s := v.(types.Set)
-
+	idx := uint64(0)
 	count := s.Len()
+	if at, ok := args[atKey].(int); ok {
+		idx = uint64(at)
+	}
 	if c, ok := args[countKey].(int); ok {
 		count = uint64(c)
 	}
 
 	// Clamp ranges
+	if idx < 0 {
+		idx = 0
+	}
+	if idx > s.Len() {
+		idx = s.Len()
```

Comment:
You can immediately return an empty set in this case. Is this stuff happy with a nil interface result (obviously if not just a single element value).

You can also immediately return in the count <= 0 case.

Like
```go
if count <= 0 || idx >= s.Len() {
  // return nil, nil ?
  return []interface{}{}, nil
}
```